src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,95 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+import { Post } from '../../types/Post';
+
+import { getAllPosts, getUserPosts } from '../../api/posts';
+
+import { Loader } from '../Loader';
+
+type Props = {
+  selectedUserId: number,
+  selectedPostId: number | null,
+  setSelectedPostId: (selectedPostId: number | null) => void,
```

Comment:
Add more specific types
```suggestion
  setSelectedPostId: React.Dispatch<React.SetStateAction<number | null>>,
```